src/hoa-renderer.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @fileOverview Omnitone HOA decoder.
+ */
+
+// Internal dependencies.
+var AudioBufferManager = require('./audiobuffer-manager.js');
+var HOARotator = require('./hoa-rotator.js');
+var HOAConvolver = require('./hoa-convolver.js');
+var Utils = require('./utils.js');
+var SystemVersion = require('./version.js');
+
+// HRIRs for optimized HOA rendering.
+// TODO(hongchan): change this with the absolute URL.
+var SH_MAXRE_HRIR_URLS = [
+  'resources/sh_hrir_o_3_ch0-ch3.wav',
+  'resources/sh_hrir_o_3_ch4-ch7.wav',
+  'resources/sh_hrir_o_3_ch8-ch11.wav',
+  'resources/sh_hrir_o_3_ch12-ch15.wav'];
```

Comment:
Chrome can decode up to 8ch. Would it be possible to have two 8 channel files instead of 4 files?